### PR TITLE
build,protobuf: stop building unnecessary protobuf

### PR DESCRIPTION
**Pull request by alan-mas**
Based on dev-inf#56 we need to avoid protobuf compiler to run for Java, Obj-C and C# everytime we build as no one is using these languages anymore.

Release note: None

### CODE REVIEW

**Review by knz — APPROVED** [review]

**Comment by knz** [comment]
thank you!
can you merge this yourself, or do you need assistance?

**Comment by alan-mas** [comment]
@knz thanks to you!
Before merge I do have something I want to clarify and maybe you can help me on here a bit (I hope):
- I can see that `master` branch is too behind from the current commit, the last updated branch (and the one from we are taking the HASH in our main repo) is `delete_example_3.9` which has a commit from Ricky that's is why I sent my PR to the branch, but that is the wat we are working on this repository? or I am missing something?

And last thing I am not able to merge it as I do not have write access to this repo =)

**Comment by knz** [comment]
This is a good question, and I recommend we continue that discussion on this new issue I've created for the occasion: https://github.com/cockroachdb/cockroach/issues/64651

I'm going to merge this PR as-is, but we probably really want to upgrade protobuf to the latest version and then apply your patch on top of that.

**Pull request merged**
